rotkehlchen/chain/evm/decoding/hop/decoder.py

```diff
@@ -0,0 +1,311 @@
+            except DeserializationError:
+                chain_label = ''
+
+        target_str = ''
+        if recipient and sender and recipient != sender:
+            target_str = f'at address {recipient} '
+
+        return f'Bridge {amount} {asset.symbol_or_name()} {chain_label}{target_str}via Hop protocol'  # noqa: E501
+
+    def _decode_withdrawal_bonded(self, context: DecoderContext) -> DecodingOutput:
+        """This function is used to decode the WithdrawalBonded events on Hop protocol."""
+        if not self.base.is_tracked(hex_or_bytes_to_address(context.transaction.input_data[4:36])):
+            return DEFAULT_DECODING_OUTPUT
+
+        if (bridge := self.bridges.get(context.tx_log.address)) is None:
+            return DEFAULT_DECODING_OUTPUT
+
+        for event in context.decoded_events:
+            if (
+                event.address in (bridge.amm_wrapper, bridge.address) and
+                event.asset.identifier == bridge.identifier and
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.BRIDGE
+                event.notes = self._get_bridge_note(event.balance.amount, event.asset)
+                event.counterparty = CPT_HOP
+                break
+
+        else:
+            asset = Asset(bridge.identifier)
+            amount = hex_or_bytes_to_int(context.transaction.input_data[36:68])
+            bonder_fee = hex_or_bytes_to_int(context.transaction.input_data[100:132])
+            norm_amount = token_normalized_value(
+                token_amount=amount - bonder_fee, token=asset.resolve_to_evm_token(),
+            )
+            action_item = ActionItem(
+                action='transform',
+                from_event_type=HistoryEventType.RECEIVE,
+                from_event_subtype=HistoryEventSubType.NONE,
+                asset=asset,
+                to_event_type=HistoryEventType.WITHDRAWAL,
+                to_event_subtype=HistoryEventSubType.BRIDGE,
+                to_notes=self._get_bridge_note(norm_amount, asset),
+                to_counterparty=CPT_HOP,
+            )
+            return DecodingOutput(action_items=[action_item])
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_transfer_sent(self, context: DecoderContext) -> DecodingOutput:
+        """This function is used to decode the TransferSentToL2 events on Hop protocol."""
+        if not self.base.is_tracked(recipient := hex_or_bytes_to_address(context.tx_log.topics[3])):  # noqa: E501
+            return DEFAULT_DECODING_OUTPUT
+
+        if (bridge := self.bridges.get(context.tx_log.address)) is None:
+            return DEFAULT_DECODING_OUTPUT
+
+        amount_raw = hex_or_bytes_to_int(context.tx_log.data[:32])
+        amount = self._get_bridgeasset_amount(amount_raw, bridge)
```

Comment:
please use kwargs. Generally if the function takes more than 1 argument always use kwargs.

Why?

1. Readability
2. Protects from:
    - Giving arguments in the wrong order. With kwargs order does not matter
    - Renaming an argument is caught early on by the python interpreter itself